typeshed/__init__.py

```diff
@@ -0,0 +1,59 @@
+import os
+import subprocess
+
+
+__all__ = ['__location__', '__version__']
+
+
+def is_git_repo(dir):
```

Comment:
Why is all this git stuff here?

Reply:
Mypy has a very similar thing. There are a few reasons to want this:
* we want a "dirty" check for building a wheel for PyPI
* we want to know the exact commit that a release was built from and publish it on PyPI, too (I'm doing that via `project_urls`)

Reply:
I still don't think it should be in the "typeshed" package. It's a private API used only by setup.py. So I think all this code might as well move into setup.py (the amount of code stays constant and the isolation improves).

(That still feels wrong: setup.py is not just the thing you use to build a wheel -- it's the thing you use to install from any other source as well. If you want to have additional assurances that you're not uploading a wheel built from a dirty working directory, ideally you should have a separate helper script. But I don't have an answer for how to make the source_url specific to a commit in that case, so I'll tolerate it in setup.py. But I still don't think it belongs in `typeshed/__init__.py`.)